src/Chapter2.hs

```diff
@@ -842,7 +856,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate x lst
+  | x < 0 || lst == [] = []
+  | otherwise = drop x (take (length lst + x) (cycle lst))
```

Comment:
Awesome solution! Though, it does a little bit more work when the given number is big and the list is small. You will cycle through the list a lot. You can optimise this with the usage of `mod` :slightly_smiling_face: